src/java/org/apache/cassandra/db/partitions/AbstractBTreePartition.java

```diff
@@ -205,12 +202,17 @@ private RowAndDeletionMergeIterator merge(Iterator<Row> rowIter, Iterator<RangeT
                                                canHaveShadowedData());
     }
 
-    private abstract class AbstractIterator extends AbstractUnfilteredRowIterator
+    private class SlicesIterator extends AbstractUnfilteredRowIterator
     {
         final Holder current;
         final ColumnFilter selection;
 
-        private AbstractIterator(Holder current, Row staticRow, ColumnFilter selection, boolean isReversed)
```

Comment:
This removed `AbstractIterator`, without any change, could be the base class for both `SlicesIterator` and the new `ClusteringsIterator`, so we can avoid some duplication, especially the long comment about fetched columns.

Reply:
good idea.